lib/router.js

```diff
@@ -79,9 +79,10 @@ Cherrytree.prototype.map = function (routes) {
 
   function eachBranch (node, memo, fn) {
     node.routes.forEach(function (route) {
-      if (!route.routes || route.routes.length === 0) {
+      if (!route.options || route.options.abstract !== true) {
```

Comment:
I'll try this version in my app an the example apps I have and see how it feels. Overall, looks like it could indeed be a simplification, especially judging by the fact that there are way less docs.